docs/root/intro/life_of_a_request.rst

```diff
@@ -0,0 +1,181 @@
+Sockets. We focus on a single Envoy proxy that is part of one of these network topologies
+below.
+
+Configuration
+-------------
+
+Envoy is a platform with a high degreee of extensibility. This results in a combinatorial
+explosion of possible request paths, depending on:
+
+* L3/4 protocol, e.g. TCP, UDP, Unix Domain Sockets.
+* L7 protocol, e.g. HTTP/1, HTTP/2, HTTP/3, gRPC, Thrift, Dubbo, Kafka, Redis and various databases.
+* Transport socket, e.g. plain text, TLS, ALTS.
+* Routing, e.g. PROXY protocol, original destination, dynamic forwarding.
+* Authentication and authorization.
+* Circuit breakers and outlier detection.
+* Many other networking, HTTP, listener, access logging, health checking, tracing and stats
+  extensions and configuration options.
+
+It's helpful to limit the nature of the configuration and request to provide an illustrative
+narrative below, so we opt for the following parameters:
+
+* An HTTP/2 request with TLS over a TCP connection for both downstream and upstream.
+* The HTTP connection manager as the only network filter.
+* The router filter as the only HTTP filter (TODO: should we add more filters?).
+* Filesystem access logging.
+* Statsd sink.
+
+We assume a static bootstrap configuration, but this is a control plane detail that is
+orthogonal to the data plane:
+
+.. code-block:: yaml
+
+  todo: simple bootstrap example of above
+
+High level architecture
+-----------------------
+
+The request processing path in Envoy has responsibilities split across two major loosely
+coupled subsystems:
+
+* *ListenerManager* where the downstream client related aspects of request processing take place.
+  This includes all processing from when the first bytes are read from the socket until a
+  decoded HTTP request is ready to forward to an upstream cluster. This subsystem is also
+  responsible for the response path to the client, the downstream HTTP/2 codec lives here.
+* *ClusterManager* where the HTTP request is forwarded to an upstream backend. This is where
+  knowledge of cluster and endpoint health, load balancing and connection pooling exists. The
+  upstream HTTP/2 codec lives here.
+
+The two subsystems are bridged with the HTTP router filter.
+
+We use the labels *ListenerManager* and *ClusterManager* above to refer to the family of modules and
+instance classes that are rooted in these top-level manager classes. In reality, there are numerous
+components that we discuss below that are instantiated prior to and during the course of a request
+by these management systems, for example listeners, filter chains, codecs, connection pools and load
+balancing data structures.
+
+.. note::
+  TODO: add architecture diagram.
+
+Request path
+------------
+
+Overview
+~~~~~~~~
+
+The request path for the initial bytes that constitute the request headers takes the following
+steps:
+
+1. A TCP connection from a downstream is accepted by an Envoy listener associated with a worker
+   thread.
```

Comment:
maybe worth a server arch paragrah, calling out that each connection will land on a single thread ("worker thread") and stay there for the lifetime of all traffic on that connection.
Threading model is one of the first questions folks tend to ask when picking up Envoy

Reply:
Yep, added a paragraph under the high-level arch section.